dotnet/src/Connectors/Connectors.Google/GeminiThinkingConfig .cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Connectors.Google;
+
+/// <summary>
+/// GeminiThinkingConfig class
+/// </summary>
+public class GeminiThinkingConfig
+{
+    private int? _thinkingBudget;
+
+    /// <summary>
+    /// The thinking budget parameter gives the model guidance on how many thinking tokens it can use for its thinking process.
+    /// A greater number of tokens is typically associated with more detailed thinking, which is needed for solving more complex tasks.
+    /// thinkingBudget must be an integer in the range 0 to 24576. Setting the thinking budget to 0 disables thinking.
+    /// Budgets from 1 to 1024 tokens will be set to 1024.
+    /// **This parameter is specific to Gemini 2.5 and similar experimental models.**
+    /// </summary>
+    [JsonPropertyName("thinkingBudget")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public int? ThinkingBudget
+    {
+        get => this._thinkingBudget;
+        set
+        {
+            if (value is < 0 or > 24576)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), "ThinkingBudget must be an integer in the range 0 to 24576.");
+            }
+            this._thinkingBudget = value == 0 ? 0 : (value < 1024 ? 1024 : value);
```

Comment:
Same as above, lets avoid any validation at our side, leave this to the server side.